code/_onclick/hud/ghost.dm

```diff
@@ -44,6 +44,25 @@
 	var/mob/dead/observer/G = usr
 	G.open_spawners_menu()
 
+/// Skyrat change START
+/obj/screen/ghost/eventsignup
+	name = "Event Signup"
+	icon_state = "eventsignupoff"
+
+/obj/screen/ghost/eventsignup/New()
+	if(check_rights(R_ADMIN, FALSE))
+		icon_state = "eventsignupon"
+
+/obj/screen/ghost/eventsignup/Click()
+	var/mob/dead/observer/G = usr
+	G.open_event_menu(src)
+
+/obj/screen/ghost/eventsignup/proc/icon_update()
+	icon_state = "eventsignup" + (usr.client.prefs.event_participation ? "on" : "off")
+	update_icon()
+		
+/// Skyrat change END
```

Comment:
Why insert this into a non-modular file, when it can easily be straight copypasted elsewhere.